server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -228,15 +232,35 @@ func (ds *cachedMysql) TeamAgentOptions(ctx context.Context, teamID uint) (*json
 	return agentOptions, nil
 }
 
+func (ds *cachedMysql) TeamFeatures(ctx context.Context, teamID uint) (*fleet.Features, error) {
+	key := fmt.Sprintf(teamFeaturesKey, teamID)
+	if x, found := ds.c.Get(key); found {
+		if features, ok := x.(*fleet.Features); ok {
```

Comment:
We might as well panic if it's not a `*fleet.Features` (a code bug) vs. silently ignoring the entry and querying the `ds.Datastore`.

Reply:
here I'm following the pattern of all the other methods in `cached_mysql`, I'm hesitant to make this method behave differently, but I agree with you.

what do you think about creating an issue to refactor accordingly?

Reply:
+1, let's do it on a separate PR/issue.